src/components/SchemaViewer.svelte

```diff
@@ -11,15 +11,29 @@
       .split(" ")
       .filter((t) => t.length > 0);
 
-    const addVisibility = (node) => {
+    const addVisibility = (node, parentNodeNames = ["__root__"]) => {
       let modifiedNode = node;
+      let parentNames = [...parentNodeNames];
 
       if (modifiedNode.fields) {
-        modifiedNode.fields.forEach(addVisibility);
+        parentNames = modifiedNode.parentNames
+          ? `${modifiedNode.parentNames}.${modifiedNode.name}`
+          : modifiedNode.name;
```

Comment:
This can be simplified to something like the following. This was part of one of your earlier revision of the PR.

```js
        parentNames = `${modifiedNode.parentNames.slice(-1)}.${modifiedNode.name}`
```